docs/src/visualization.md

```diff
@@ -118,6 +118,30 @@ When plotted together with the mesh, this will yield the following visualization
 
 ![plot-rho-uniform-mesh](https://user-images.githubusercontent.com/72009492/130951039-f2f91760-539a-4e96-ac39-4053e934040b.PNG)
 
+## Plotting a user-defined scalar field
+
+To plot a scalar quantity, one can call `plot(ScalarPlotData2D(u, semi))`, where
+`u` is an array of nodal values of the scalar field to plot. The layout of `u` should
+match the layout of the `x` and `y` nodal coordinates of the respective solver. For
+example, after running
+`trixi_include(joinpath("examples", "unstructured_2d_dgsem", "elixir_euler_wall_bc.jl"))`,
+the following can be used to plot the function `f(x, y) = x * y`:
+```julia
+x = view(semi.cache.elements.node_coordinates, 1, :, :, :)
+y = view(semi.cache.elements.node_coordinates, 2, :, :, :)
+plot(ScalarPlotData2D(x .* y, semi))
+```
+This produces the following plot:
+
+![scalar-plotting-example](https://user-images.githubusercontent.com/1156048/132762371-da141802-34e8-4035-a88d-4d60e66c9f19.png)
+
+This routine can be used to visualized scalar quantities which depend on the solution,
+such as the norm of a velocity vector or two-dimensional vorticity.
+
+!!! note
+    When visualizing a scalar field, the plotted solution is reinterpolated using a
+    high order polynomial approximation. Thus, small discrepancies may be observed when
+    the underlying data is highly non-smooth or under-resolved.
 
```

Comment:
Do you have a (somewhat clean) example for plotting the vorticity, say for some `DGMulti` solver?

Reply:
Sure - how's the following? This script should work for any `DGMulti` solver for the compressible Euler equations.

```julia
compute_vorticity(velocity, semi) = 
    compute_vorticity(velocity, Trixi.mesh_equations_solver_cache(semi)...)

function compute_vorticity(velocity, mesh, equations::CompressibleEulerEquations2D, dg::DGMulti, cache)
    rd = dg.basis
    md = mesh.md

    @unpack Dr, Ds = rd
    @unpack rxJ, sxJ, ryJ, syJ, J = md

    v1, v2 = velocity
    dv1dy = ryJ .* (Dr * v1) + syJ .*(Ds * v1)
    dv2dx = rxJ .* (Dr * v2) + sxJ .*(Ds * v2)
    return dv2dx - dv1dy
end

rho, rhou, rhov, E = StructArrays.components(sol.u[end])
v1 = rhou ./ rho
v2 = rhov ./ rho
vorticity = compute_vorticity((v1, v2), semi)
plot(ScalarPlotData2D(vorticity, semi; variable_name="Vorticity at t = $(tspan[end])"))
```

Reply:
It would be great if you could add this as an example to the docs, too! Maybe even as a (not too expensive) doctest?

Reply:
Sure. How do you add a doctest?

Reply:
Nvm - found the example in the Jacobian code.

Reply:
Our current tutorials have some examples, e.g., https://raw.githubusercontent.com/trixi-framework/Trixi.jl/main/docs/src/differentiable_programming.md. The documentation is available at https://juliadocs.github.io/Documenter.jl/stable/man/doctests/. The idea behind this is to make sure that the syntax doesn't change accidentally, leaving dead code in the docs.

Reply:
Feel free to ping me if you need help with that.